### DISCUSSION TITLE
How to get stream by selecting stream profile from axis camera using axis sdk?

### DISCUSSION THREAD

**Discussion by maazullah96** [AxisCommunications/acap3-examples#205] [General] [answered]
I'm working on a project that requires me to retrieve a stream from a specific stream profile. However, I couldn't find any relevant functions or modules that could help me with this task. Can you guide me on how I can retrieve a stream from a specific stream profile?

**Comment by pataxis** [comment]
Hi @maazullah96 , thanks for your question.

To take out a stream through an ACAP application we have the [vdostream](https://github.com/AxisCommunications/acap3-examples/tree/main/vdostream) example. To retrieve a stream from a specific stream profile I would have to refer to the [technical documentation](https://www.axis.com/techsup/developer_doc/acap3/3.5/api/vdostream/html/index.html) of the [Video Capture API](https://help.axis.com/en-us/acap-3-developer-guide#video-capture-api) if it's possible. I found one function [vdo_channel_get_stream_profile()](https://www.axis.com/techsup/developer_doc/acap3/3.5/api/vdostream/html/vdo-channel_8h.html#a69e3ce84ef7bf204417794aeb8651ce0) that sounds interesting.

For doing this from a server or something similar I would recommend you to search in the [VAPIX documentation](https://www.axis.com/vapix-library/).

**Reply by maazullah96** [reply]
THanks @pataxis 

I tried using vdo_channel_get_stream_profile() but I am getting this error.

I could not find any proper documentation on how to use these functions. can you  suggest me what i should do.

` Failed to get stream profile: DBus call failed: Profile unavailable`

The camera I am using contains these video stream profiles
![Screenshot from 2023-05-03 18-17-17](https://user-images.githubusercontent.com/30190807/235929317-04af50be-0818-4fa2-aac0-dd6c40e4d797.png)

Source code I am using  
```c
for (GList *iter = channels; iter != NULL; iter = iter->next){
        VdoChannel *channel_ = VDO_CHANNEL(iter->data);
       gchar* profile_name = g_strdup("h264-demo");
        printf("Channel ID: %d\n", vdo_channel_get_id(channel_));
        VdoMap *info = vdo_channel_get_info(channel_, &error);
        guint value = vdo_map_get_uint32(info, "name", 0);
        printf("name: %d\n", vdo_channel_get_id(channel_));
        syslog(LOG_INFO, " name: %d \n", value);
        VdoMap *profile = vdo_channel_get_stream_profile(channel_, profile_name, VDO_FORMAT_H264, &error);
        if (error != NULL) {
            syslog(LOG_ERR, "%s %d \n", error->message,vdo_channel_get_id(channel_));
            printf("Error: %s \n", error->message);
            // g_error_free(error);
            printf("\n");
                // continue;
        }
              printf("\n");
    }
```

**Reply by pataxis** [reply, marked as answer]
Hi again, the code looked correct to me and I checked with the API team, they said that,

> Currently only the "default" profile is implemented

Unfortunately this limitation is not documented.
